code/modules/tgui/states/physical.dm

```diff
@@ -0,0 +1,24 @@
+ /**
+  * tgui state: physical_state
+  *
+  * Short-circuits the default state to only check physical distance.
+ **/
+
+GLOBAL_DATUM_INIT(tgui_physical_state, /datum/tgui_state/physical, new)
+
+/datum/tgui_state/physical/can_use_topic(src_object, mob/user)
+	. = user.shared_tgui_interaction(src_object)
+	if(. > STATUS_CLOSE)
+		return min(., user.physical_can_use_tgui_topic(src_object))
+
+/mob/proc/physical_can_use_tgui_topic(src_object)
+	return STATUS_CLOSE
+
+/mob/living/physical_can_use_tgui_topic(src_object)
+	return shared_living_tgui_distance(src_object)
+
+/mob/living/silicon/physical_can_use_tgui_topic(src_object)
+	return max(STATUS_UPDATE, shared_living_tgui_distance(src_object)) // Silicons can always see.
+
+/mob/living/silicon/ai/physical_can_use_tgui_topic(src_object)
+	return STATUS_UPDATE // AIs are not physical.
```

Comment:
no obscured state one?

Reply:
Will implement

Reply:
UIs currently close when the AI loses vision of them anyway

Reply:
guessing this is related to the new cameranet system and obscured system TG has.